inc/Engine/Media/Embeds/EmbedsSubscriber.php

```diff
@@ -0,0 +1,237 @@
+<?php
+namespace WP_Rocket\Engine\Media\Embeds;
+
+use WP_Rocket\Admin\Options_Data;
+use WP_Rocket\Event_Management\Subscriber_Interface;
+use WP_Rocket\ThirdParty\ReturnTypesTrait;
+use function rocket_bypass;
+
+/**
+ * Event subscriber to control Embeds behavior.
+ *
+ * @since  3.7 Moved to new architecture.
+ * @since  3.2
+ */
+class EmbedsSubscriber implements Subscriber_Interface {
+	use ReturnTypesTrait;
+
+	/**
+	 * The Options Data instance.
+	 *
+	 * @var Options_Data
+	 */
+	private $options;
+
+	/**
+	 * EmbedsSubscriber constructor.
+	 *
+	 * @param Options_Data $options An Options Data instance.
+	 */
+	public function __construct( Options_Data $options ) {
+		$this->options = $options;
+	}
+
+	/**
+	 * Return an array of events that this subscriber wants to listen to.
+	 *
+	 * @since  3.7
+	 * @access public
```

Comment:
```suggestion
```

We don't need the access in the docblock, it's already indicated by the `public` keyword in the method declaration